syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransform.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+import org.apache.beam.sdk.values.Row;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+public class DataGeneratorSchemaTransform implements SchemaTransform, Serializable {
+
+  private final long recordsPerSecond;
+  private final long secondsToRun;
+  private final String schema;
+
+  public DataGeneratorSchemaTransform(long recordsPerSecond, long secondsToRun, String schema) {
+    this.recordsPerSecond = recordsPerSecond;
+    this.secondsToRun = secondsToRun;
+    this.schema = schema;
+  }
+
+  @Override
+  public PTransform<PCollectionRowTuple, PCollectionRowTuple> buildTransform() {
+    return new PTransform<PCollectionRowTuple, PCollectionRowTuple>() {
+      @Override
+      public PCollectionRowTuple expand(PCollectionRowTuple input) {
+        org.apache.beam.sdk.schemas.Schema beamSchema =
+            AvroUtils.toBeamSchema(new Parser().parse(schema));
+        final Instant startTime = Instant.now();
+        PCollection<Instant> instants =
+            input
+                .getPipeline()
+                .apply(
+                    PeriodicImpulse.create()
+                        .startAt(startTime)
+                        .stopAt(Instant.now().plus(Duration.standardSeconds(secondsToRun)))
+                        .withInterval(Duration.millis(1000))
+                        .applyWindowing());
+        instants = instants.apply(Reshuffle.viaRandomKey());
+
+        PCollection rows =
+            instants
+                .apply(FlatMapElements.via(new InstantToRowFn(startTime, schema, beamSchema)))
+                .setRowSchema(beamSchema);
+        return PCollectionRowTuple.of("output", rows);
+      }
+    };
+  }
+
+  private class InstantToRowFn extends SimpleFunction<Instant, List<Row>> {
+
+    private final Instant startTime;
+
+    /**
+     * The avro schema string. This is needed to pass addition attributes like the string field size
+     * and a list of values for a field. TODO: Once implementing support for passing additional
+     * attributes in Beam Schema, avroSchemaString can be removed.
+     */
+    private final String avroSchemaString;
+
+    private final org.apache.beam.sdk.schemas.Schema beamSchema;
+
+    public InstantToRowFn(
+        Instant startTime, String avroSchemaString, org.apache.beam.sdk.schemas.Schema beamSchema) {
+      this.startTime = startTime;
+      this.avroSchemaString = avroSchemaString;
+      this.beamSchema = beamSchema;
+    }
+
+    @Override
+    public List<Row> apply(Instant input) {
+      final Schema schema = Schema.parse(this.avroSchemaString);
+      long ordinal = (input.minus(Duration.millis(startTime.getMillis())).getMillis() / 1000);
+      return LongStream.range(ordinal * recordsPerSecond, (ordinal + 1) * recordsPerSecond)
+          .mapToObj(intVal -> RecordCreator.createRowRecord(schema, beamSchema))
+          .collect(Collectors.toList());
+    }
```

Comment:
This looks complicated. Can we add a commend explain what it's doing?

Reply:
added comments and exmaples